File: 2021.06.10/MediaPlayerExample/app/src/main/java/com/example/mediaplayerexample/MainActivity.java
package com.example.mediaplayerexample;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {


    Button btn_start;
    Button btn_stop;

    MediaPlayer background;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        background = MediaPlayer.create(getBaseContext(), R.raw.music);

        btn_start = (Button) findViewById(R.id.btn_start);
        btn_start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                background = MediaPlayer.create(getBaseContext(), R.raw.music);
                background.start();
                background.setLooping(true);
            }
        });


        btn_stop = (Button) findViewById(R.id.btn_stop);
        btn_stop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                background.stop();
            }
        });
    }
}

File: 2020.10.15/RecyclerView/RecyclerView/app/src/main/java/com/recyclerview/sample/MainActivity.kt
package com.recyclerview.sample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.recyclerview.sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val boxAdapter = SampleAdapter()

        //레이아웃 매니저 지정
        binding.recyclerView.adapter = boxAdapter
        binding.recyclerView.layoutManager = LinearLayoutManager(this)

        boxAdapter.add(1)
        boxAdapter.add(2)
        boxAdapter.add(1)
        boxAdapter.add(1)
        boxAdapter.add(2)
        boxAdapter.add(1)
        boxAdapter.add(2)
        boxAdapter.add(1)
        boxAdapter.add(1)
        boxAdapter.add(2)
        boxAdapter.add(1)
        boxAdapter.add(2)
        boxAdapter.add(1)
        boxAdapter.add(1)
        boxAdapter.add(2)
    }
}